pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -120,9 +122,13 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 		return result, client.IgnoreNotFound(err)
```

Comment:
let's change the get on 120 to get it into a different object than the spc above. Makes the code read a little clearer.

maybe call it `toCleanSpc` or something similar. Using the same spc object for creation and deletion logic is a little weird and may lead to bugs in the future.

This applies to the backend reconciler and placeholder pod reconciler too